aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -136,6 +148,11 @@ func resourceAwsLambdaEventSourceMappingCreate(d *schema.ResourceData, meta inte
 		params.StartingPosition = aws.String(startingPosition.(string))
 	}
 
+	if startingPositionTimestamp, ok := d.GetOk("starting_position_timestamp"); ok {
```

Comment:
Do we also need to confirm that `starting_position` is also `lambda.EventSourcePositionAtTimestamp` or can this be specified without it?

Reply:
The API returns an error if its incorrectly set (e.g. `starting_position = "LATEST"` and `starting_position_timestamp = "..."): `InvalidParameterValueException: Timestamp only valid with StartingPosition AT_TIMESTAMP`

Since the API returns a helpful error and `CustomizeDiff` has a lot of issues for cross-attribute validation, we try to avoid the extra coding, but I can add it in there if you wish.

Reply:
No need! That makes sense to me!